src/main/java/com/dataguise/saas/DgControllerAuthProvider.java

```diff
@@ -0,0 +1,85 @@
+package com.dataguise.saas;
+
+import com.dg.saas.orch.client.DgSecureRestDriver;
+import com.dg.saas.orch.exception.DgDrvException;
+import com.dg.saas.orch.models.structures.ErrorConstants;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.*;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+
+@Component
+public class DgControllerAuthProvider implements AuthenticationProvider {
+    @Autowired
+    private DgSecureRestDriver restDriver;
+    private Logger logger = Logger.getLogger(DgControllerAuthProvider.class);
+
+    @Override
+    public Authentication authenticate(Authentication authentication)
+            throws AuthenticationException {
+
+        String name = authentication.getName();
+        Object password = authentication.getCredentials();
+
+        // Null check on object type to avoid null pointer exception on toString()
+        if (password == null && "".equals(name)) {
+            throw new AuthenticationCredentialsNotFoundException(ErrorConstants.ERROR_EMPTY_CREDENTIAL.getErrorMessage());
+        }
+
+        logger.info("Login request for name: " + name);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/DgControllerAuthProvider.java L33
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-498860664>*